code/framework/extension/src/main/java/io/cattle/platform/extension/impl/ExtensionMap.java

```diff
@@ -0,0 +1,81 @@
+package io.cattle.platform.extension.impl;
+
+import io.cattle.platform.extension.ExtensionManager;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ExtensionMap<K, V> implements Map<K, V> {
+
+    Map<K, V> map;
+    ConcurrentHashMap<K, V> inner;
+    String key;
+    ExtensionManager extensionManager;
+
+    public ExtensionMap(ExtensionManager extensionManager, String key, Map<K, V> map) {
+        this.extensionManager = extensionManager;
+        this.key = key;
+        this.inner = map == null ? new ConcurrentHashMap<K, V>() : new ConcurrentHashMap<>(map);
+        this.map = Collections.unmodifiableMap(inner);
```

Comment:
See io.cattle.platform.extension.impl.ExtensionList#ExtensionList As to why this is needed. Basically so that the injected lists using the extensionmanager.list method returns a list or map in the case of io.cattle.platform.extension.impl.ExtensionMap#ExtensionMap So that the classes that use `@inject` cannot modify the list.